pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -259,12 +262,19 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 		}
 	}
 
+	k8sEndpoint, err := utils.GetK8sServiceEndPoint(r.client)
+	if err != nil {
+		log.Error(err, "Error reading services endpoint configmap")
+		r.status.SetDegraded("Error reading services endpoint configmap", err.Error())
```

Comment:
Do you need to unset degraded after a successful retry?  (I genuinely don't know due to not knowing the operator very well.)

Reply:
No need to do that. It is cleared when every thing is ready.